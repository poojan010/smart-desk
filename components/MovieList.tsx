import React from 'react';
import { View, Text, StyleSheet, FlatList } from 'react-native';

interface Movie {
  id: string;
  title: string;
}

interface MovieListProps {
  movies: Movie[];
}

const MovieList: React.FC<MovieListProps> = ({ movies }) => (
  <View style={styles.container}>
    {movies.length > 0 ? (
      <FlatList 
        data={movies}
        renderItem={({item}) => {
          return(
            <Text key={item.id}>{item.title}</Text>
          )
        }}
      />
    ) : (
      <Text>No movies available</Text>
    )}
  </View>
);

const styles = StyleSheet.create({
  container: {
    padding: 10,
  },
});

export default MovieList;
