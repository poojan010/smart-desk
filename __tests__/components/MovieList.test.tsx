import { render, screen } from "@testing-library/react-native";

import MovieList from "@/components/MovieList";

const movies = [
    {
        id: 'movie1',
        title: "Movie One"
    },
    {
        id: 'movie2',
        title: "Movie Two"
    },
    {
        id: 'movie3',
        title: "Movie Three"
    },
    {
        id: 'movie4',
        title: "Movie Four"
    },
]

describe("<MovieList/>", () => {

    test("Snapshot", () => {
        const { toJSON } = render(<MovieList movies={movies} />)

        expect(toJSON()).toMatchSnapshot()
    })

    test("MovieList Renders correctly (with items)", () => {
        render(<MovieList movies={movies} />)

        movies.forEach((item) => {
            screen.getByText(item.title)
        })
    })

    test("MovieList Renders correctly (without items)", () => {
        render(<MovieList movies={[]} />)

        
        screen.getByText("No movies available")
        
    })


})