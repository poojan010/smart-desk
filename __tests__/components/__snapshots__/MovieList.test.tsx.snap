// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieList/> Snapshot 1`] = `
<View
  style={
    {
      "padding": 10,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "id": "movie1",
          "title": "Movie One",
        },
        {
          "id": "movie2",
          "title": "Movie Two",
        },
        {
          "id": "movie3",
          "title": "Movie Three",
        },
        {
          "id": "movie4",
          "title": "Movie Four",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text>
          Movie One
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text>
          Movie Two
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text>
          Movie Three
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text>
          Movie Four
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
